pkg/reconciler/subscription/subscription.go

```diff
@@ -436,36 +434,52 @@ func (r *Reconciler) syncPhysicalChannel(ctx context.Context, sub *v1.Subscripti
 	}
 }
 
-func (r *Reconciler) patchSubscription(ctx context.Context, namespace string, channel *eventingduckv1.Channelable, sub *v1.Subscription) (bool, error) {
-	after := channel.DeepCopy()
-
-	if sub.DeletionTimestamp.IsZero() {
-		r.updateChannelAddSubscription(after, sub)
-	} else {
-		r.updateChannelRemoveSubscription(after, sub)
-	}
-
-	patch, err := duck.CreateMergePatch(channel, after)
-	if err != nil {
-		return false, err
-	}
-	// If there is nothing to patch, we are good, just return.
-	// Empty patch is {}, hence we check for that.
-	if len(patch) <= 2 {
-		return false, nil
-	}
-
+func (r *Reconciler) updateChannelable(ctx context.Context, namespace string, channel *eventingduckv1.Channelable, sub *v1.Subscription) (bool, error) {
 	resourceClient, err := eventingduck.ResourceInterface(r.dynamicClientSet, namespace, channel.GroupVersionKind())
 	if err != nil {
 		logging.FromContext(ctx).Warnw("Failed to create dynamic resource client", zap.Error(err))
 		return false, err
 	}
-	patched, err := resourceClient.Patch(ctx, channel.GetName(), types.MergePatchType, patch, metav1.PatchOptions{})
+	uo := &unstructured.Unstructured{}
+	err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		channel, err = r.getChannel(ctx, sub)
+		if err != nil {
+			return err
+		}
+
+		after := channel.DeepCopy()
+
+		if sub.DeletionTimestamp.IsZero() {
+			r.updateChannelAddSubscription(after, sub)
+		} else {
+			r.updateChannelRemoveSubscription(after, sub)
+		}
+
+		// If there is nothing to update, we are good, just return.
+		if len(channel.Spec.Subscribers) == len(after.Spec.Subscribers) {
+			return nil
+		}
+
+		output, err := runtime.DefaultUnstructuredConverter.ToUnstructured(after)
+		if err != nil {
+			return err
+		}
+		
+		uo.Object = output
+		if _, err = resourceClient.Update(ctx, uo, metav1.UpdateOptions{}); err != nil {
+			return err
```

Comment:
A little better error message:
```suggestion
			return fmt.Errorf("failed to update channel %s/%s: %w", uo.GetNamespace(), uo.GetName(), err)
```